app/src/main/java/ua/dancecamp/vmedvediev/ua21summerdancecamp/UI/splash/SplashActivity.kt

```diff
@@ -0,0 +1,123 @@
+import ua.dancecamp.vmedvediev.ua21summerdancecamp.model.ApplicationSettings
+import ua.dancecamp.vmedvediev.ua21summerdancecamp.model.Event
+import ua.dancecamp.vmedvediev.ua21summerdancecamp.model.EventsCache
+import ua.dancecamp.vmedvediev.ua21summerdancecamp.repository.Repository
+import java.io.IOException
+import java.lang.ref.WeakReference
+import java.nio.charset.Charset
+import java.util.*
+
+class SplashActivity : AppCompatActivity() {
+
+    private val splashViewModel by lazy {
+        ViewModelProviders.of(this, SplashViewModel(Repository(RealmEventMapper(), RealmSettingsMapper())).SplashViewModelFactory())
+                .get(SplashViewModel::class.java)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_splash)
+        splashViewModel.apply {
+            getApplicationSettings()
+            applicationSettings.observe(this@SplashActivity, Observer {
+                if (it != null) {
+                    val localeLanguage = baseContext.resources.configuration.locale.language
+                    val applicationLanguage = if (it.interfaceLanguage.isEmpty()) {
+                        localeLanguage
+                    } else {
+                        it.interfaceLanguage
+                    }
+                    val previousLocaleLanguage = if (it.localeLanguage.isEmpty()) {
+                        localeLanguage
+                    } else {
+                        it.localeLanguage
+                    }
+                    when {
+                        EventsCache.eventsList.isNotEmpty() && (previousLocaleLanguage == applicationLanguage) -> {
+                            startActivity(Router.prepareMainActivityIntent(this@SplashActivity))
+                            finish()
+                        }
+                        splashViewModel.getEvents().isNotEmpty() && (previousLocaleLanguage == applicationLanguage) -> {
+                            splashViewModel.setupLocalStorage()
+                            startActivity(Router.prepareMainActivityIntent(this@SplashActivity))
+                            finish()
+                        }
+                        else -> JsonParserAsyncTask(WeakReference(this@SplashActivity)).execute(applicationLanguage)
+                    }
+                    updateLocale(applicationLanguage)
+                    splashViewModel.saveApplicationSettigns(ApplicationSettings(it.id, applicationLanguage, applicationLanguage))
+                }
+            })
+        }
+    }
+
+    private fun updateLocale(applicationLanguage: String) {
+        val locale = Locale(applicationLanguage)
+        Locale.setDefault(locale)
+        val config = baseContext.resources.configuration
+        config.setLocale(locale)
+        baseContext.resources.updateConfiguration(config, baseContext.resources.displayMetrics)
```

Comment:
`baseContext ` ? Why?

Reply:
I reviewed examples on the google.com "how to update locale in Android"